server/src/services/library.service.ts

```diff
@@ -359,6 +364,10 @@ export class LibraryService {
       }
     }
 
+    if (dto.isReadOnly !== undefined) {
+      await this.assetRepository.setReadOnlyForLibrary(id, dto.isReadOnly);
```

Comment:
I think you can just use `updateAll` for this.

Reply:
I believe that `updateAll` requires ids of the items to update, so first I'd need to do a query to get a list of all the ids for the library and then I'd need to do a second query to update them all. This does the update in a single query, which is much easier in my view

Reply:
A dedicated repo method probably makes the most sense.

Reply:
Oh, I might have misremembered how it works. I'm fine with a separate repo method.